featuretools/computational_backends/pandas_backend.py

```diff
@@ -331,6 +331,46 @@ def strip_values_if_series(values):
 
         return frame
 
+    def _calculate_groupby_features(self, features, entity_frames):
+        entity_id = features[0].entity.id
+        assert len(set([f.entity.id for f in features])) == 1, \
+            "features must share base entity"
+        assert entity_id in entity_frames
+
+        frame = entity_frames[entity_id]
+
+        for f in features:
+            # handle when no data
+            if frame.shape[0] == 0:
+                set_default_column(frame, f)
+                continue
+
+            groupby = f.groupby.get_name()
```

Comment:
if we make the change to feature tree, we can ensure that all features have the same groupby once they get to this step. this means we can do the iterations of `for index, group in frame_data.groupby(groupby)` only once per group and calculate all the features at once